rust/src/lib.rs

```diff
@@ -1864,13 +1712,255 @@ pub unsafe extern "C" fn zcashlc_rewind_to_height(
     unwrap_exc_or(res, false)
 }
 
+/// A struct that contains a subtree root.
+///
+/// # Safety
+///
+/// - `root_hash_ptr` must be non-null and must be valid for reads for `root_hash_ptr_len`
+///   bytes, and it must have an alignment of `1`.
+/// - The total size `root_hash_ptr_len` of the slice pointed to by `root_hash_ptr` must
+///   be no larger than `isize::MAX`. See the safety documentation of `pointer::offset`.
+#[repr(C)]
+pub struct FfiSubtreeRoot {
```

Comment:
suggestion: check that other structs have the FFI all in caps.

Reply:
I followed Rust making convention here, where acronyms are CamelCased (`HtmlFoo` instead of `HTMLFoo`).

Reply:
I assumed so! It's kind of a philosophical questions, should the FFI target swift generator's conventions or Rust's? 

I don't have a preference, I'd just pick one or the other.